x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/EsqlActionTaskIT.java

```diff
@@ -216,9 +222,14 @@ private ActionFuture<EsqlQueryResponse> startEsql() {
         return startEsql("from test | stats sum(pause_me)");
     }
 
+    /**
+     * Start an ESQL query, releasing a few docs from the {@code pause_me}
+     * script so it'll actually start but won't finish it's first page.
+     */
     private ActionFuture<EsqlQueryResponse> startEsql(String query) {
         scriptPermits.drainPermits();
-        scriptPermits.release(between(1, 5));
+        prereleasedDocs = between(1, 5);
```

Comment:
The number 5 here seems a bit magical. Can we express this in terms of `pageSize` so it's clear that this never covers all pages?

Reply:
YEah!